### PR TITLE
merge bitcoin#16097: Add Flags enum to ArgsManager class

### PR DESCRIPTION
**Pull request by kittywhiskers**

### CODE REVIEW

File: src/bench/bench_dash.cpp [hunk 1]
```diff
@@ -20,16 +20,16 @@ static const char* DEFAULT_BENCH_FILTER = ".*";
 
 static void SetupBenchArgs()
 {
-    gArgs.AddArg("-?", "Print this help message and exit", false, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-list", "List benchmarks without executing them", false, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-filter=<regex>", strprintf("Regular expression filter to select benchmark by name (default: %s)", DEFAULT_BENCH_FILTER), false, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-asymptote=n1,n2,n3,...", strprintf("Test asymptotic growth of the runtime of an algorithm, if supported by the benchmark"), false, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-output_csv=<output.csv>", "Generate CSV file with the most important benchmark results.", false, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-output_json=<output.json>", "Generate JSON file with all benchmark results.", false, OptionsCategory::OPTIONS);
+    gArgs.AddArg("-?", "Print this help message and exit", ArgsManager::ALLOW_ANY, false, OptionsCategory::OPTIONS);
```
Comment (PastaPastaPasta) [inline src/bench/bench_dash.cpp]: for `Use Flags enum in AddArg() ` you should've just used the scripted diff..

File: src/bench/bench_dash.cpp [hunk 2]
```diff
@@ -20,16 +20,16 @@ static const char* DEFAULT_BENCH_FILTER = ".*";
 
 static void SetupBenchArgs()
 {
-    gArgs.AddArg("-?", "Print this help message and exit", ArgsManager::ALLOW_ANY, false, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-list", "List benchmarks without executing them", ArgsManager::ALLOW_ANY, false, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-filter=<regex>", strprintf("Regular expression filter to select benchmark by name (default: %s)", DEFAULT_BENCH_FILTER), ArgsManager::ALLOW_ANY, false, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-asymptote=n1,n2,n3,...", strprintf("Test asymptotic growth of the runtime of an algorithm, if supported by the benchmark"), ArgsManager::ALLOW_ANY, false, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-output_csv=<output.csv>", "Generate CSV file with the most important benchmark results.", ArgsManager::ALLOW_ANY, false, OptionsCategory::OPTIONS);
-    gArgs.AddArg("-output_json=<output.json>", "Generate JSON file with all benchmark results.", ArgsManager::ALLOW_ANY, false, OptionsCategory::OPTIONS);
+    gArgs.AddArg("-?", "Print this help message and exit", ArgsManager::ALLOW_ANY, OptionsCategory::OPTIONS);
```
Comment (PastaPastaPasta) [inline src/bench/bench_dash.cpp:23]: should use scripted diff `Use ArgsManager::DEBUG_ONLY flag`

**Review by PastaPastaPasta — COMMENTED** [review]
LGTM, just change to be scripted diffs

**Review by PastaPastaPasta — DISMISSED** [review]
utACK for merging via sqash commit

**Comment by UdjinM6** [comment]
LGTM but shouldn't this be merged via squashing @PastaPastaPasta ?

**Comment by UdjinM6** [comment]
And one more thing - let's avoid naming commits "merge bitcoin#xxxx: ..." in future PRs when all of them are from the same original PR and the PR is meant for squashing. This is confusing to review and it also creates a mess in the original PR spamming it with tons of links.

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**